foodsaving/users/api.py

```diff
@@ -46,8 +46,18 @@ def conversation(self, request, pk=None):
         return self.retrieve_private_conversation(request, pk)
 
     def get_queryset(self):
-        return self.queryset.filter(Q(groups__in=self.request.user.groups.all()) |
-                                    Q(id=self.request.user.id)).distinct()
+        is_member_of_group = Q(groups__in=self.request.user.groups.all())
+
+        is_self = Q(id=self.request.user.id)
+
+        group_ids = self.request.user.groups.values_list('id', flat=True)
+        is_applicant_of_group = Q(groupapplication__group__id__in=group_ids)
```

Comment:
Can you also write this instead?

```
groups = self.request.user.groups.all()
is_applicant_of_group = Q(groupapplication__group__in=groups)
```

From my understanding, it should produce the same SQL with less verbose Python.
But could be that you tried and it didn't work for some reason?

Reply:
Ah, yes, that looks better! I will give it a shot.